src/app/system/manage-surveys/create-survey/create-survey.component.html

```diff
@@ -0,0 +1,161 @@
+              <mat-error *ngIf="surveyForm.controls.key.hasError('required')">
+                Key is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+            
+            <mat-form-field fxFlex="48%">
+              <mat-label>Name</mat-label>
+              <input matInput maxlength="255" required formControlName="name">
+              <mat-error *ngIf="surveyForm.controls.name.hasError('required')">
+                Name is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field fxFlex="48%">
+              <mat-label>Country Code</mat-label>
+              <input matInput maxlength="2" required formControlName="countryCode">
+              <mat-error *ngIf="surveyForm.controls.countryCode.hasError('required')">
+                Country Code is <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="surveyForm.controls.countryCode.hasError('pattern')">
+                Country Code <strong>must consist of 2 uppercase characters</strong>
+              </mat-error>
+            </mat-form-field>
+
+            <mat-form-field fxFlex="98%">
+              <mat-label>Description</mat-label>
+              <textarea matInput formControlName="description"></textarea>
+            </mat-form-field>
+
+            <div fxFlexFill fxLayout="row" fxLayout.xs="column" fxLayoutAlign="center" fxLayoutGap="5px" *ngIf="!questionDatas.length">
+              <button type="button" mat-raised-button [routerLink]="['../']">Cancel</button>
+              <button type="button" mat-raised-button color="primary" [disabled]="!(this.surveyForm.get('key').valid && this.surveyForm.get('name').valid && this.surveyForm.get('countryCode').valid)" (click)="addQuestion()">
+                Proceed
+              </button>
+            </div>
+
+          </div>
+
+        </mat-card-content>
+
+      </mat-card>
+
+      <mat-card formArrayName="questionDatas" *ngFor="let question of questionDatas.controls; let questionIndex = index; last as isLast">
+
+        <mat-card-content>
+
+          <div fxFlexFill fxLayoutGap="2%" fxLayout.lt-md="column" fxLayout="row wrap" [formGroupName]="questionIndex">
+
+            <div fxFlex="98%" fxLayout="row wrap" fxLayoutGap="2%" fxLayoutAlign="space-between center">
+              <h2 class="mat-h2">Question {{ questionIndex + 1 }}</h2>
+              <button mat-raised-button color="warn" (click)="removeQuestion(questionIndex)" [disabled]="questionDatas.controls.length === 1">
+                <fa-icon icon="trash"></fa-icon>&nbsp;&nbsp;
+                Delete Question
+              </button>
+            </div>
+
+            <mat-form-field fxFlex="48%">
+              <mat-label>Key</mat-label>
+              <input matInput maxlength="32" required formControlName="key">
+              <mat-error *ngIf="question.controls.key.hasError('required')">
```

Comment:
I am aware that in other components, this would be questionsData.at(questionIndex)..., but I do not see the point, as this is much more readable, and would not require the get function to keep running over and over again.